src/utils/attributes.ts

```diff
@@ -0,0 +1,68 @@
+import { createObserver } from "./observers";
+
+type AttributeObject = { [k: string]: any };
+
+/**
+ * Watches global attributes of a component.
+ *
+ * Derived from: https://gist.github.com/willmartian/b4dd6b57d71dd0438fb9e7c6f4048578
+ */
+export interface WatchedComponent {
+  /**
+   * The host element.
+   */
+  readonly el: HTMLElement;
+
+  /**
+   * The watched attributes object.
+   * Should be stateful.
+   * '@State() inheritedAttributes = {};'
+   */
+  inheritedAttributes: AttributeObject;
+
+  /**
+   * The MutationObserver to listen to inherited attributes.
+   */
+  inheritedAttributesObserver: MutationObserver;
+}
+
+/**
+ * Helper to set up listening for changes to global attributes.
+ *
+ * render() {
+ *   const lang = this.inheritedAttributes['lang'] ?? 'en';
+ *   return <div>My lang is {lang}</div>;
+ * }
+ */
+export function watchAttributes(component: WatchedComponent, attributes: string[]): void {
+  const attributeObject: AttributeObject = {};
+  const { el } = component;
+
+  const updateAttributesObject = () => {
+    attributes.forEach((attr) => {
+      if (el.hasAttribute(attr)) {
+        const value = el.getAttribute(attr);
+        if (value !== null) {
+          attributeObject[attr] = el.getAttribute(attr);
+        }
+      }
+    });
+
+    component.inheritedAttributes = { ...attributeObject };
+  };
+
+  updateAttributesObject();
+
+  component.inheritedAttributesObserver = createObserver("mutation", () => updateAttributesObject());
```

Comment:
Can this utility use a shared observer? It would make it simpler on implementors since they wouldn't need to store anything.

Reply:
Can you explain this one? I'm not seeing it.

Reply:
Shared meaning you pass in your own observer to the `watchGlobalAttributes` function?

Reply:
I meant having the utility use and manage the observer itself. This would eliminate implementors having to add a `globalAttributesObserver` property on a component.

Reply:
The only bummer about that is if the component forgets to disconnect the observer it stays in the utility class.

Reply:
True, but this would be a problem for any utils/patterns we have.